bookkeeper-common/src/main/java/org/apache/bookkeeper/common/util/OrderedExecutor.java

```diff
@@ -348,7 +348,11 @@ public ExecutorService chooseThread(Object orderingKey) {
             return threads[0];
         }
 
-        return threads[MathUtils.signSafeMod(orderingKey.hashCode(), threads.length)];
+        if (null == orderingKey) {
+            return threads[rand.nextInt(threads.length)];
+        } else {
+            return threads[MathUtils.signSafeMod(orderingKey.hashCode(), threads.length)];
+        }
```

Comment:
Looks fine, but wondering what is the usecase of this? I worry it might mask bugs as ordered safe executor is an absolute must to protect the correctness (wherever it is needed). So callers need to know what they are doing.

Reply:
it is for convenience, so in the caller, I don't have to do `if (orderKey != null) chooseThread(orderKey); else chooseThread(..)`. I can have a separate util function to do that, but I feel it is better to do it at the executor level.